lua/kznllm/presets.lua

```diff
@@ -237,7 +237,7 @@ function M.invoke_llm(make_data_fn, make_curl_args_fn, make_job_fn, opts)
       opts.debug_fn(data, M.NS_ID, stream_end_extmark_id, opts)
     else
       local _, crow, ccol = unpack(vim.fn.getpos '.')
-      stream_end_extmark_id = api.nvim_buf_set_extmark(M.BUFFER_STATE.ORIGIN, M.NS_ID, crow - 1, ccol, { strict = false })
+      stream_end_extmark_id = api.nvim_buf_set_extmark(M.BUFFER_STATE.ORIGIN, M.NS_ID, crow - 1, ccol - 1, { strict = false })
```

Comment:
This should solve it

From the docs: https://neovim.io/doc/user/api.html#nvim_buf_set_extmark()

> end_col : ending col of the mark, 0-based exclusive.

Feels like this happens way too often because `vim.fn.getpos '.'` gives 1-indexed values, while `vim.api.nvim_buf_set_extmark` expects 0-indexed params...

I was somewhat aware of this, but didn't notice the col was exclusive so I need to adjust it by 1. 🥲

---

For context about lua concepts, this is the only function that we use to output streamed text to the buffer:

```
---@param content string
---@param extmark_id integer
function M.write_content_at_extmark(content, ns_id, extmark_id)
  local extmark = api.nvim_buf_get_extmark_by_id(0, ns_id, extmark_id, { details = false })
  local mrow, mcol = extmark[1], extmark[2]

  local lines = vim.split(content, '\n')

  vim.cmd 'undojoin'
  api.nvim_buf_set_text(0, mrow, mcol, mrow, mcol, lines)
end
```

We pass an `extmark` that automatically updates and moves its position when you insert text on/before the position

_side note: the other place I would've checked is the `vim.api.nvim_buf_set_text` in `M.write_content_at_extmark`, but it would be much more noticeable_